### PR TITLE
RCNConfigFetch: fix weakSelf/strongSelf usage

### LINKED ISSUE #6123: RCNFetch.m line 246 Random crash after updating the SDK to 6.26.0
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.4.1
  * Firebase SDK version: 6.26.0
  * Firebase Component: "Crashlytics", "Analytics", "Auth", "RemoteConfig", "Database", "Performance"
  * Installation method: CocoaPods

### [REQUIRED] Step 2: Describe the problem

Random crash after upgrading the SDK to 6.26.0

#### Steps to reproduce:

The following crash happens on low network on app launch or app update.

```
Crashed: com.apple.main-thread
0  ******                        0x106dabb94 __65-[RCNConfigFetch refreshInstallationsTokenWithCompletionHandler:]_block_invoke_2 + 246 (RCNFetch.m:246)
1  ******                         0x106d2f964 __49-[FIRInstallations installationIDWithCompletion:]_block_invoke + 181 (FIRInstallations.m:181)
2  FBLPromises                    0x10805b9f8 __56-[FBLPromise chainOnQueue:chainedFulfill:chainedReject:]_block_invoke.48 + 273 (FBLPromise.m:273)
3  libdispatch.dylib              0x190f0e9a8 _dispatch_call_block_and_release + 24
4  libdispatch.dylib              0x190f0f524 _dispatch_client_callout + 16
5  libdispatch.dylib              0x190ec1640 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1044
6  CoreFoundation                 0x1911c7748 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
7  CoreFoundation                 0x1911c261c __CFRunLoopRun + 1724
8  CoreFoundation                 0x1911c1c34 CFRunLoopRunSpecific + 424
9  GraphicsServices               0x19b30b38c GSEventRunModal + 160
10 UIKitCore                      0x1952f422c UIApplicationMain + 1932
11 *********                        0x1046cd180 main + 17 (AccessibilityIdentifiers.swift:17)
12 libdyld.dylib                  0x191049800 start + 4

```

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Fixes #6123.

### CODE REVIEW

File: FirebaseRemoteConfig/Sources/RCNFetch.m [hunk 1]
```diff
@@ -476,11 +490,12 @@ - (void)fetchWithUserProperties:(NSDictionary *)userProperties
 
       // We had a successful fetch. Update the current eTag in settings if different.
       NSString *latestETag = ((NSHTTPURLResponse *)response).allHeaderFields[kETagHeaderName];
-      if (!self->_settings.lastETag || !([self->_settings.lastETag isEqualToString:latestETag])) {
-        self->_settings.lastETag = latestETag;
+      if (!strongSelf->_settings.lastETag ||
+          !([self->_settings.lastETag isEqualToString:latestETag])) {
```
Comment (ncooke3) [inline FirebaseRemoteConfig/Sources/RCNFetch.m:494]: This looks good! Just noticed this though, should the `self` here be a `strongSelf`?

Comment (maksymmalyhin) [reply FirebaseRemoteConfig/Sources/RCNFetch.m:494]: Good catch! Yeah, ideally this code should be refactored to improve readability.

**Comment by maksymmalyhin** [comment]
> Thanks! Is a unit test feasible?

Feasible, but it appeared that unit testing is not set up for the class. I'll try to some quick tests for these specific cases. Also I updated #6132 with tests action item.

**Comment by maksymmalyhin** [comment]
As discussed offline, the tests will have to be added separately.

**Pull request merged**